### DISCUSSION TITLE
Great concept but barely usable

### DISCUSSION THREAD

**Discussion by doobydooby72** [celzero/rethink-app#1142] [Q&A]
This is a broad scope of issues I've encountered over the past day or so.

> Wireguard working at 1st and then not working at all and then unable to turn off wireguard specifically.

> DNS Sometimes working with my VPN's DNS but mostly not working.

> RethinkDNS crashing every time i selected your DNS after changing.

> DNS leaking when using wireguard (still experimenting with DNS' but this occured even when i did get my VPN's DNS and Wireguard config working. 1 or 2 occasions for a few minutes where DNS wasnt leaking)

> No other DNS options showing under DoH, crypt or proxy after clearing storage, this went along with the crashes if i selected RDNS. I couldnt resolve this issue even with reinstalling, i had to fall back to the previous release to get it not to crash and show other DNS options at this point.

> DNS leak when just running wireguard and the firewall, no in app DNS function. (This seems relevant as i would've thought the system DNS option {which also seems odd} or disabling the apps DNS function would fallback to wireguards hooked DNS since thats what the official wireguard seems to do and doesn't get DNS leaks)

Device: Lenovo tablet running android 9.0

Great app in concept but barely works in my limited time with it, look forward to it advancing. Cant really recommend in its current state though (unless its just my luck). The fork of openvpn for personalDNSfilter worked better when i last used it over a year ago even though it was quite slow.

**Comment by ignoramous** [comment]
> Wireguard working at 1st and then not working at all and then unable to turn off wireguard specifically.

Without more information, I can only guess you're hitting #1015 which we have since fixed but not yet released.

> DNS Sometimes working with my VPN's DNS but mostly not working.

WireGuard DNS is NOT used in the current versions; but we've fixed that too #1049 

> RethinkDNS crashing every time i selected your DNS after changing.

Can you please email us the "bug report", if you're comfortable doing so? See top half of the "About" page on the app.

> DNS leaking when using wireguard (still experimenting with DNS' but this occured even when i did get my VPN's DNS and Wireguard config working. 1 or 2 occasions for a few minutes where DNS wasnt leaking)

Strange because WireGuard DNS is simply not used in the current version. But this has now been implemented for the newer releases (see above).

> No other DNS options showing under DoH, crypt or proxy after clearing storage, this went along with the crashes if i selected RDNS. 

Hm, seems like an issue of database corruption. A bug report (as mentioned above, would have helped), if the app hadn't been re-installed. But that's okay.

(I am turning this into a discussion, since it isn't reporting any one particular "issue" for us to fix; let me know if you've more suggestions / questions :)

**Reply by onetimecontributor** [reply]
should this be closed? nothing of it makes sense as of 2025

**Reply by ignoramous** [reply]
I tend to leave the conversations open if the OP has not replied or not marked any reply as answer... But you're probably right that all issues pointed out here have been fixed in `v055o`, which we're releasing only on GitHub on Aug 1.

**Reply by xa4hf8** [reply]
I am eager to try the pre-release versions because some of my bug reports & troubleshooting tasks are probably going to be obsolete now... but i think the [web site](https://rethinkdns.com/app) should say _**"use [Obtainium](https://obtainium.imranr.dev/) for beta/pre-release builds"**_ or something similar.

For your convenience, here is a download badge of the same size as the others.

<img width="441" height="168" alt="OBT-pink neon 168" src="https://github.com/user-attachments/assets/86f4f845-3b28-4126-bb05-08776fe622b6" />
<img width="441" height="168" alt="OBT-blue neon 168" src="https://github.com/user-attachments/assets/e16b0513-22a2-495e-a86a-fb4c549b91d9" />

**Reply by ignoramous** [reply]
I don't use Obtainium but I see very many do. What should "Get it on Obtanium" link to for Rethink? GitHub releases page? Or, Obtanium has a special deep link / url?

**Reply by onetimecontributor** [reply]
Described here I guess https://github.com/ImranR98/apps.obtainium.imranr.dev/blob/main/README.md#url-redirection

Just to test it: 
https://apps.obtainium.imranr.dev/redirect.html?r=obtainium://add/https://github.com/celzero/rethink-app

UPD: It looks like httpS link works (it opens Obtainium at least), but as I have already installed rdnsapp via Obtainium, can't test it clean atm.
I have posted the simplest form without any complications described in that guide. As it's only 1 apk in assets, it should be ok I think

**Reply by xa4hf8** [reply]
In the Obtainium _add app_ dialog, note that `verify the 'latest'` switch may prevent detection of pre-releases, so its probably best to use a pre-configured JSON link, with the standard Obtaininum download badge for production releases and the experimental badge for the beta channel

**Reply by onetimecontributor** [reply]
> In the Obtainium _add app_ dialog, note that `verify the 'latest'` switch may prevent detection of pre-releases, so its probably best to use a pre-configured JSON link, with the standard Obtaininum download badge for production releases and the experimental badge for the beta channel

can you please share the working snippet/json/link?

**Reply by xa4hf8** [reply]
- JSON

`{"id":"com.celzero.bravedns","url":"https://github.com/celzero/rethink-app","author":"celzero","name":"Rethink","additionalSettings":"{\"includePrereleases\":true}"}`
___
- URL

https://apps.obtainium.imranr.dev/redirect.html?r=obtainium://app/%7B%22id%22%3A%22com.celzero.bravedns%22%2C%22url%22%3A%22https%3A%2F%2Fgithub.com%2Fcelzero%2Frethink-app%22%2C%22author%22%3A%22celzero%22%2C%22name%22%3A%22Rethink%22%2C%22additionalSettings%22%3A%22%7B%5C%22includePrereleases%5C%22%3Atrue%7D%22%7D

**Reply by onetimecontributor** [reply]
@ignoramous , for "stable" version, official badge is https://github.com/ImranR98/Obtainium/blob/main/assets/graphics/badge_obtainium.png and the link is https://apps.obtainium.imranr.dev/redirect.html?r=obtainium://add/https://github.com/celzero/rethink-app

For "pre-release" version, the link is in the comment above, and the badge is ... there is no official one I think, but something is posted here https://github.com/celzero/rethink-app/discussions/1142#discussioncomment-14083501

**Reply by xa4hf8** [reply]
😉 with all due respect, the _"stable"_ releases are _"anything but"_ on my platform, and i dont understand the reason for marking these new builds as `pre-release` when reverting to a previous build is supported by F-Droid, Obtainium, _et cetera_ in the rare case that it would be needed.

**Reply by onetimecontributor** [reply]
> 😉 with all due respect, the _"stable"_ releases are _"anything but"_ on my platform, and i dont understand the reason for marking these new builds as `pre-release` when reverting to a previous build is supported by F-Droid, Obtainium, _et cetera_ in the rare case that it would be needed.

It's your experience, I'm still on N and it works good for me unless I'm missing smth
I consider to upgrade one of devices to latest pre-release.
So it's wise to keep new ones as pre-release
Disclaimer: I'm not a dev of rdns app

**Reply by DwainZwerg** [reply]
For pre-releases there is the "Include prereleases“-Toggle. If deactivated (default behavior) pre-releases are not suggested. If activated pre-releases are suggested.

**Reply by onetimecontributor** [reply]
> For pre-releases there is the "Include prereleases“-Toggle. If deactivated (default behavior) pre-releases are not suggested. If activated pre-releases are suggested.

yep, it was asked for a separate badge here https://github.com/celzero/rethink-app/discussions/1142#discussioncomment-14083501

**Reply by doobydooby72** [reply]
Apologies everyone, feel free to close if you want. I did try all of the updates and botched my phones install and reverted to 55n... 55n has been the most stable with the odd freeze, force stop, start again... All the updates after leaked DNS consistently with my wiredguard VPN config... Still happy to see gradual progress. 

I haven't tried this on my ancient tablet in a while maybe I'll give it a shot soon and see how it goes

**Reply by ignoramous** [reply]
Added Obtainium to the Readme.
